admin_ui/app/role/RoleEditPage.js

```diff
@@ -1,12 +1,129 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import { Link } from '@reach/router';
+import { useObservable } from 'rxjs-hooks';
 import useDocumentTitle from '@rehooks/document-title';
+import Store from '../Store';
+import Input from '../../components/Input';
+import Textarea from '../../components/Textarea';
+import Button from '../../components/Button';
+import Select from 'react-select';
+import RoleDeleteModal from './RoleDeleteModal';
 
 const RoleEditPage = ({ roleId }) => {
   useDocumentTitle(`Edit role#${roleId}`);
+  // Load the store
+  const store = React.useContext(Store);
+  // Establish the values for the changedPermissions$ and changedOrg$ flags
+  // (we'll use to decide whether to show "Save changes" buttons in the appropriate sections)
+  const changedPermissions = useObservable(
+    () => store.role.changedPermissions$,
+    store.role.changedPermissions$.getValue()
+  );
+  const changedOrg = useObservable(() => store.role.changedOrg$, store.role.changedOrg$.getValue());
```

Comment:
You need to combine this observable with the one above to avoid multiple renders. Here's how to do this...

```
// in addition to you current imports...
import { combineLatest } from 'rxjs';
import { debounceTime } from 'rxjs/operators';

const [changedPermissions, changedOrg] = useObservable(
  () => combineLatest(store.role.changedPermissions$, store.role.changedOrg$).pipe(debounceTime(0)),
  [store.role.changedPermissions$.getValue(), store.role.changedOrg$.getValue()]
);
```

You combine both streams in an array, use debounceTime to ensure both streams have been updated before `useObservable` triggers and finally you pass some default values.